flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/sink/writer/config/BasicAsyncSinkWriterConfiguration.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    public int getMaxBatchSize() {
+        return maxBatchSize;
+    }
+
+    public int getMaxInFlightRequests() {
+        return maxInFlightRequests;
+    }
+
+    public int getMaxBufferedRequests() {
+        return maxBufferedRequests;
+    }
+
+    public long getMaxBatchSizeInBytes() {
+        return maxBatchSizeInBytes;
+    }
+
+    public long getMaxTimeInBufferMS() {
+        return maxTimeInBufferMS;
+    }
+
+    public long getMaxRecordSizeInBytes() {
+        return maxRecordSizeInBytes;
+    }
+
+    public RateLimitingStrategy getRateLimitingStrategy() {
+        return rateLimitingStrategy;
+    }
+
+    public static BasicAsyncSinkWriterConfigurationBuilder builder() {
+        return new BasicAsyncSinkWriterConfigurationBuilder();
+    }
+
+    /** Builder for {@link BasicAsyncSinkWriterConfiguration}. */
+    public static class BasicAsyncSinkWriterConfigurationBuilder {
+
+        private int maxBatchSize = 500;
+        private int maxInFlightRequests = 50;
+        private int maxBufferedRequests = 10_000;
+        private long maxBatchSizeInBytes = 5 * 1024 * 1024;
+        private long maxTimeInBufferMS = 5_000;
+        private long maxRecordSizeInBytes = 1024 * 1024;
+        private RateLimitingStrategy rateLimitingStrategy;
+
+        public BasicAsyncSinkWriterConfigurationBuilder setMaxBatchSize(int maxBatchSize) {
+            this.maxBatchSize = maxBatchSize;
+            return this;
+        }
+
+        public BasicAsyncSinkWriterConfigurationBuilder setMaxInFlightRequests(
+                int maxInFlightRequests) {
+            this.maxInFlightRequests = maxInFlightRequests;
+            return this;
+        }
+
+        public BasicAsyncSinkWriterConfigurationBuilder setMaxBufferedRequests(
+                int maxBufferedRequests) {
+            this.maxBufferedRequests = maxBufferedRequests;
+            return this;
+        }
+
+        public BasicAsyncSinkWriterConfigurationBuilder setMaxBatchSizeInBytes(
+                long maxBatchSizeInBytes) {
+            this.maxBatchSizeInBytes = maxBatchSizeInBytes;
+            return this;
+        }
+
+        public BasicAsyncSinkWriterConfigurationBuilder setMaxTimeInBufferMS(
+                long maxTimeInBufferMS) {
+            this.maxTimeInBufferMS = maxTimeInBufferMS;
+            return this;
+        }
+
+        public BasicAsyncSinkWriterConfigurationBuilder setMaxRecordSizeInBytes(
+                long maxRecordSizeInBytes) {
+            this.maxRecordSizeInBytes = maxRecordSizeInBytes;
+            return this;
+        }
+
+        public BasicAsyncSinkWriterConfigurationBuilder setRateLimitingStrategy(
+                RateLimitingStrategy rateLimitingStrategy) {
+            this.rateLimitingStrategy = rateLimitingStrategy;
+            return this;
+        }
+
+        public BasicAsyncSinkWriterConfiguration build() {
+            if (rateLimitingStrategy == null) {
```

Comment:
What about if we do not want a rate limiting strategy? Should the sink use a no-op rather than null?